packages/ra-ui-materialui/src/form/FormInput.tsx

```diff
@@ -29,7 +29,9 @@ const FormInput = <RecordType extends Record | Omit<Record, 'id'> = Record>(
 ) => {
     const { input, classes: classesOverride, ...rest } = props;
     const classes = useStyles(props);
-    const { id, ...inputProps } = input ? input.props : { id: undefined };
+    const { id, className, ...inputProps } = input
+        ? input.props
+        : { id: undefined, className: undefined };
```

Comment:
Using an empty object will do the trick. No need to assign "undefined" since things are already "undefined" in the empty object

Reply:
TypeScript disagree